pkg/kubelet/cm/dra/state/state_checkpoint.go

```diff
@@ -20,54 +20,24 @@ import (
 	"fmt"
 	"sync"
 
-	"k8s.io/apimachinery/pkg/types"
-	"k8s.io/apimachinery/pkg/util/sets"
 	"k8s.io/kubernetes/pkg/kubelet/checkpointmanager"
 	"k8s.io/kubernetes/pkg/kubelet/checkpointmanager/errors"
+	"k8s.io/kubernetes/pkg/kubelet/cm/dra/state/checkpoint/v1alpha1"
 )
 
 var _ CheckpointState = &stateCheckpoint{}
 
-// CheckpointState interface provides to get and store state
 type CheckpointState interface {
 	GetOrCreate() (ClaimInfoStateList, error)
 	Store(ClaimInfoStateList) error
 }
 
-// ClaimInfoState is used to store claim info state in a checkpoint
-// +k8s:deepcopy-gen=true
-type ClaimInfoState struct {
-	// ClaimUID is an UID of the resource claim
-	ClaimUID types.UID
+type ClaimInfoStateList []v1alpha1.ClaimInfoState
 
-	// ClaimName is a name of the resource claim
-	ClaimName string
-
-	// Namespace is a claim namespace
-	Namespace string
-
-	// PodUIDs is a set of pod UIDs that reference a resource
-	PodUIDs sets.Set[string]
-
-	// DriverState contains information about all drivers which have allocation
-	// results in the claim, even if they don't provide devices for their results.
-	DriverState map[string]DriverState
-}
-
-// DriverState is used to store per-device claim info state in a checkpoint
-// +k8s:deepcopy-gen=true
-type DriverState struct {
-	Devices []Device
-}
-
-// Device is how a DRA driver described an allocated device in a claim
-// to kubelet. RequestName and CDI device IDs are optional.
-// +k8s:deepcopy-gen=true
-type Device struct {
-	PoolName     string
-	DeviceName   string
-	RequestNames []string
-	CDIDeviceIDs []string
+type DRACheckpoint interface {
+	checkpointmanager.Checkpoint
+	GetData() ClaimInfoStateList
+	SetData(ClaimInfoStateList)
 }
```

Comment:
What is this interface for? I don't see it used / implemented anywhere...

Reply:
This is a leftover from the previous implementation. Will remove it, thanks for pointing out.